src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -556,17 +572,40 @@ private[fastq] case class SampleInfo(sample: Sample, isUnmatched: Boolean = fals
 private[fastq] object FastqDemultiplexer {
 
   /** Stores the minimal information for a single template read. */
-  case class DemuxRecord(name: String, bases: String, quals: String, molecularBarcode: Option[String],
+  final case class DemuxRecord(name: String, bases: String, quals: String, molecularBarcode: Option[String],
                          sampleBarcode: Option[String], readNumber: Int, pairedEnd: Boolean, comment: Option[String],
-                         originalBases: Option[String] = None, originalQuals: Option[String] = None)
+                         originalBases: Option[String] = None, originalQuals: Option[String] = None) {
+
+    /** A function to mask bases that have a quality score less than or equal to a specified threshold
+      * @param threshold The threshold for masking bases, inclusive. Bases with a quality score <= threshold will be masked
+      * @param qualityEncoding The encoding used for quality scores in the Fastq file.
+      * @return a new DemuxRecord with updated bases
+      */
+    def maskLowQualityBases(threshold: Int, qualityEncoding: QualityEncoding): DemuxRecord = {
+      val bases = this.bases.zip(quals).map { case (base, qual) =>
+        if (qualityEncoding.toStandardNumeric(ch = qual) <= threshold) "N" else base
+      }
+    this.copy(bases = bases.mkString)
+    }
+  }
 
   /** A class to store the [[SampleInfo]] and associated demultiplexed [[DemuxRecord]]s.
     * @param sampleInfo the [[SampleInfo]] for the matched sample.
     * @param numMismatches the # of mismatches if it matched a sample with a sample barcode.  This will be [[Int.MaxValue]]
     *                      for the unmatched sample.
     * @param records the records, one for each read that has template bases.
     */
-  case class DemuxResult(sampleInfo: SampleInfo, numMismatches: Int, records: Seq[DemuxRecord])
+  final case class DemuxResult(sampleInfo: SampleInfo, numMismatches: Int, records: Seq[DemuxRecord]) {
+
+    /** A function to mask bases that have a quality score less than or equal to a specified threshold
+      * @param threshold The threshold for masking bases, inclusive. Bases with a quality score <= threshold will be masked
+      * @return a new DemuxResult with updated bases
+      */
+    def maskLowQualityBases(threshold: Int, qualityEncoding: QualityEncoding): DemuxResult = { // using this.type here causes a mismatch error
+      if (threshold == 0) this
```

Comment:
```suggestion
      if (threshold <= 0) this
```
Just to be defensive.

Reply:
Ah, here's where we check if `threshold` is leq zero.  Not a bad idea to also add the check in `DemuxRecord`